spec/services/activity_defaults_spec.rb

```diff
@@ -9,6 +9,11 @@
   let(:project) { create(:project_activity, :gcrf_funded, parent: programme) }
   let(:third_party_project) { create(:third_party_project_activity, :gcrf_funded, parent: project) }
 
+  let(:ispf_oda_programme) { create(:programme_activity, :ispf_funded) }
+  let(:ispf_non_oda_programme) { create(:programme_activity, :ispf_funded, is_oda: false) }
+  let(:ispf_oda_project) { create(:project_activity, parent: ispf_oda_programme, is_oda: true) }
+  let(:ispf_non_oda_project) { create(:project_activity, parent: ispf_non_oda_programme, is_oda: false) }
```

Comment:
Without manually setting `is_oda` here the third party projects end up with `nil` as their `is_oda` value. My intuition is that this is because `activity_defaults` is only invoked in the UI user journey flow, so when manually creating a project in RSpec the defaults aren't added. Does that sound correct? If so, we'll need to work out how to make this behaviour the default in bulk upload as well, so I wonder if there's somewhere we can set even more default (baked in) parent-based defaults

Reply:
As discussed on slack:
> `activity_defaults` is called in `new_child` which is currently called by the (current) bulk importer, as well as the `activity_children_controller` so it should set those values by default. When we create an activity in the specs we're using the factory, so that method isn't called.

This means we should be covered when we do get around to bulk importing these activities